### ISSUE TITLE
JS error on serializing non ASCII into URL

### ISSUE THREAD

**Issue by seancolsen** [mathesar-foundation/mathesar#2666]
## Steps to reproduce

1. Go to the table page
1. Add a filter condition on a text column
1. Enter the following text into the value field:

    > ## foo

1. Observe that Mathesar appends the following query param to the URL

    `?q=W1sxLDUwMF0sW10sW10sWyJhbmQiLFtbMiwiZXF1YWwiLCJmb28iXV1dXQ`

    This is good.

1. Now change filter value to:

    > ## 💩

1. Observe a JS error: 

    > DOMException: String contains an invalid character

## Cause

1. In `Url64.ts` we're using `btoa` and `atob`.

1. It turns out these functions [choke when given Unicode](https://developer.mozilla.org/en-US/docs/Web/API/btoa#unicode_strings).

    > # 🥴

## Other effects

- This bug originally got reported as a schema page that wouldn't load any tables. It turns out one table had a _column name_ with non-unicode characters and we were trying to create a link to the exploration page for that table by serializing the entire table object (including some column data) into the URL. I have opened #2667 to offer so marginal improvement in that scenario.

## Possible solutions

- MDN has some [crazy examples](https://developer.mozilla.org/en-US/docs/Glossary/Base64#solution_2_%E2%80%93_rewriting_atob_and_btoa_using_typedarrays_and_utf-8) of ways to deal with this problem

- I'm inclined to search for a reliable third-party lib that can handle this for us

**Comment by shenderaj9021** [comment]
@seancolsen I want work on this issue can you please assign it to me ?

**Comment by seancolsen** [comment]
No @shenderaj9021, this issue is already assigned. See our [Contributor Guide](https://wiki.mathesar.org/en/community/contributing).

**Comment by pavish** [comment]
@seancolsen 

For 0.1.1, I'd like to remove base64 encoding of the URL entirely. I don't see a good enough reason to have it, the change in url length isn't a major concern compared to the issues we need to handle to perform base64 encoding/decoding in JS.

**Comment by perguth** [comment]
This also happens if you e.g. set `Formatting` -> `Currency Symbol` -> `€`.

**Comment by seancolsen** [comment]
@perguth thanks for reporting that. I just tried the steps I _think_ you're describing but I didn't observe any errors. Can you open a new ticket describing the reproduction steps in more detail. This might be an unrelated issue.

**Comment by kgodey** [comment]
@seancolsen I was able to reproduce @perguth's issue on our staging server by setting up a Money type column with the euro currency symbol. The table loads fine at first, but if you navigate away, then the schema page doesn't load https://staging.mathesar.org/mathesar_tables/5/

Staging is running `master`, so this PR's changes are not yet deployed there.

**Comment by seancolsen** [comment]
Thanks for that info @kgodey. It looks like I didn't try correctly/hard-enough. I can reproduce this that money-symbol bug locally with the master branch. When I previously tried reproducing this, I was using the develop branch, naively thinking that because #2680 had not yet been merged I would have a decent simulacrum of the installation where @perguth noticed the issue. Alas, I should have used `master` instead of `develop` because it appears that #2667 most likely fixed the money symbol issue, and that had already been merged into `develop` by the time I attempted to reproduce.

Now that #2680 is also merged, I think we're doubly covered here. I checked out `master` reproduced the bug. Then checked out `develop` and was not able to reproduce it.
